packages/api/src/routers/api/ai.ts

```diff
@@ -0,0 +1,401 @@
+- If the user is requesting for a specific set of data (ex. "frontend service", filter the condition by the appropriate property based on the below properties ex. \`ServiceName = 'frontend'\`)
+- Pick an appropriate time range based on the user's request if one can be inferred from the request.
+
+The user is looking to do a query on their data source named: ${source.name} of type ${source.kind}.
+
+The ${source.kind === SourceKind.Log ? 'log level' : 'span status code'} is stored in ${source.severityTextExpression}.
+You can identify services via ${source.serviceNameExpression}
+${
+  source.kind === SourceKind.Trace
+    ? `Duration of spans can be queried via ${source.durationExpression} which is expressed in 10^-${source.durationPrecision} seconds of precision.
+Span names under ${source.spanNameExpression} and span kinds under ${source.spanKindExpression}`
+    : `The log body can be queried via ${source.bodyExpression}`
+}
+Various log/span-specific attributes as a Map can be found under ${source.eventAttributesExpression} while resource attributes that follow the OpenTelemetry semantic convention can be found under ${source.resourceAttributesExpression}
+You must use the full field name ex. "column['key']" or "column.key" as it appears.
+
+The following is a list of properties and example values that exist in the source:
+${JSON.stringify(keyValues)}
+
+There may be additional properties that you can use as well:
+${JSON.stringify(allFieldsWithKeys.slice(0, 200).map(f => ({ field: f.key, type: f.type })))}
+`;
+
+      logger.info(prompt);
+
+      const result = await generateObject({
+        model,
+        schema: z.object({
+          displayType: z.enum([DisplayType.Line, DisplayType.Table]),
+          select: z
+            .array(
+              // @ts-ignore
+              z.object({
+                // TODO: Change percentile to fixed functions
+                aggregationFunction: AggregateFunctionSchema.describe(
+                  'SQL-like function to aggregate the property by',
+                ),
+                property: z
+                  .string()
+                  .describe(
+                    'Property or column to be aggregated (ex. Duration)',
+                  ),
+                condition: z
+                  .string()
+                  .optional()
+                  .describe(
+                    "SQL filter condition to filter on ex. `SeverityText = 'error'`",
+                  ),
+              }),
+            )
+            .describe('Array of data series or columns to chart for the user'),
+          groupBy: z
+            .string()
+            .optional()
+            .describe('Group by column or properties for the chart'),
+          timeRange: z
+            .string()
+            .default('Past 1h')
+            .describe(
+              'Time range of data to query for like "Past 1h", "Past 24h"',
+            ),
+        }),
+        prompt,
```

Comment:
For the demo it's not a big deal - but we should try and pass in `system` and `prompt` separately.. the `system` flag shouldn't have anything user facing and has higher priority resulting in decreased risk of prompt injection